# [Title]

## 1. Introduction

### 1.1 Literature review of the compound (<500 words)

**β-carotene** 

β-carotene belongs to the large group of caroterpernoids, also called tetraterpenoids. Caroterpernoids are organic red/orange pigments. They are found naturally in many plants and fungi as well as some algae and bacteria. β-carotene are well known for giving plants as carrots and pumkin their characteristic orange color [1]. 

As Caroterpernoids are tetraterpenoids they are all synthesized from 8 isoprene molecules and have the chemical formula $C_{40}$$H_{x}$. Caroterpernoids devided in two classes xanthophylls and carotenes. Xanthophylls contains oxygen whereas carotenes is purely carbohydrates. β-carotene ($C_{40}$$H_{56}$) belongs in the carotene class as can be seen on the figure below [1]. 

![Beta-carotine drawing](Figures/Beta-Carotin.png)

β-carotene has a number of biological functions. It serves as a precursor for vitamin A, which is related to a number of important functions in the human body including proper growth and function of sight[1,2]. Moreover, it is shown to have properties as antioxidant and anticancer.

β-carotene is a valuble propduct within the food, pharmalogical and cosmetic indutry. Not only for its bioactive properties but also as an additive for colouring in food and pharmaceutical tablets. Within cosmetics it is often used in creams as it protects the skin against oxidation and UV radiation[2]. 

**Current production methods**



----------------------
[1] Berman, Judit, et al. “Nutritionally Important Carotenoids as Consumer Products.” Phytochemistry Reviews, vol. 14, no. 5, Springer Netherlands, 2015, pp. 727–43, doi:10.1007/s11101-014-9373-1, 10.1007/11101.1572-980X.
[2] https://academic.oup.com/fqs/article/2/2/69/4970186

**Current production methods**

- Extracted from plants.
- Chemical synthesis 
- Biosynthesis

### 1.2 Literature review of the cell factory (<500 words)
*Saccharomyces cerevisiae* was chosen as the cell factory for this project, as it is a well-known organism with beneficial features in industrial production setups. It differs from the model bacterial organism *Escherichia coli* by its larger size and is presence of nucleus, and gives other possibilities of production that are not available in bacteria[1]

*S. cerevisiae* is a eukaryotic single-celled organism of the species yeast, under the phylum ascomycetes. It has been used for thousands of years in various productions like wine, brewing and cheese making, and is considered non-pathogenic, which emphasises its importance to many nations. Due to the age of usage it has been studied as a model eukaryote for many years and was the first eukaryotic cell with the full genome sequenced[1]. 

As there have been made extensive work, which has created thorough knowledge of the specie, many advanced molecular biological tools for genetically manipulation are available. This makes it suitable to work with in the laboratory, to test new pathways and products. This is not let alone with the fact that it is advanced in growth speed and is well established in industrial biotechnology, making it suitable as a cell factory and widely used in the fermentation-based industry[2]. 

As this project focuses on beta-carotene production, *S. cerevisiae* has proven to be even more relevant. As conventional yeast, *S. cerevisiae* can produce geranylgeranyl diphosphate, which is a precursor in the carotenoid biosynthesis. Through genetic engineering it has been proven possible to produce beta-carotenes in *S. cerevisiae*. Compared to other heterlogous beta-carotene production organisms, it is considered safe which is important, as beta-carotenes can be used in food production etc.[4]

Due to the extensive knowledge of *S. cerevisiae* and that it has already been proven possible to produce beta-carotene in the organism, we found it relevant to take a computational look at this organism and available gene sequence models (GSM). This will hopefully make it possible to predict higher production titers and better hypothetical genetic engineering strategies for production optimisation. More specificly, we will work with the strain *S. cerevisiae S288C*[3], which has two GSMs available on BIGG.com*


[1]Madigan, M. T., Martinko, J. M., & Parker, J. (2014). Brock biology of microorganisms. Upper Saddle River, NJ: Prentice Hall/Pearson Education.
[2]Li, Q., Sun, Z., Li, J., & Zhang, Y. (2013). Enhancing beta-carotene production in Saccharomyces cerevisiae by metabolic engineering. FEMS microbiology letters, 345(2), 94–101. https://doi.org/10.1111/1574-6968.12187
[3] Fisk, D. G., Ball, C. A., Dolinski, K., Engel, S. R., Hong, E. L., Issel-Tarver, L., Schwartz, K., Sethuraman, A., Botstein, D., Cherry, J. M., & Saccharomyces Genome Database Project (2006). Saccharomyces cerevisiae S288C genome annotation: a working hypothesis. Yeast (Chichester, England), 23(12), 857–865. https://doi.org/10.1002/yea.1400

[4] Qian Li, Zhiqiang Sun, Jing Li, Yansheng Zhang, Enhancing beta-carotene production in Saccharomyces cerevisiae by metabolic engineering, FEMS Microbiology Letters, Volume 345, Issue 2, August 2013, Pages 94–101, https://doi.org/10.1111/1574-6968.12187

* This statement is based on seach on the website http://bigg.ucsd.edu/search?query=Saccharomyces+cerevisiae the 10/11/2020.



## 2. Problem definition (<300 words)
Beta-carotene is a carotenoid with numerous biological functions in the human body including being involved in the natural synthesis of provitamin A, among others necessary for growth, vision and embryotic development. Beta-carotene is furthermore widely used in the industry as a coloring agent in food and cosmetics and an additive in medicine as it has shown anticancer as well as antioxidant properties. Despite the numerous functions of beta-carotene, we are not able to synthesize it in the human body, pushing market demands for its synthetic production. 
Chemical synthesis is as of today the most commonly used technique for its production, but research has revealed that synthetic beta-carotene has carcinogenic activity and effective production methods for its natural synthesis is thus of great interest. Based on this extensive research has been caried out to examine the microbial biosynthesis of it as it provides a solid basis for a natural and sustainable production while avoiding the use of organic solvents such as heavy metals, strong acids and bases along with the formation of byproducts. Metabolic engineering has additionally carved the way to improve the yield and thus obtain higher titers and productivity of the product, through strain engineering. All though utilizing a natural producer of beta-carotene seem like the straightforward choice as production organism, it has been found that most of these are non-cultivable in laboratory conditions and that they require extensive optimalization. Choosing a well characterized microorganism, with various tools available for genetic engineering will thus enhance the chances and ease the process of increasing the product yield and productivity of the strain. 
We have utilized an existing genome scale metabolic model (GSM) for *S. cerevisiae* and incorporate relevant reactions and pathways for its natural synthesis of beta-carotene. Afterwards we have performed phenotype predictions based on gene deletions and essentiality and evaluated on metabolic pathway visualizations with Escher.

Kiler artilker: 
β-Carotene—properties and production methods (Food Quality and Safety)
Engineering microbial factories for synthesis of value-added products (J Ind Microbiol Biotechnol.)

## 3. *If Project category II:* Selection and assessment of existing GSM (<500 words)

Using the data base BiGG, two GSM models for *S. cerevisias* were identified. Both are from the same strain, strain S288C, however iMM904, contains 1226 metabolites, 1557 reaction and 905 genes while iND750 is a bit smaller with 1059 metabolites, 1266 reactions and 750 genes. Before begining on the engineering strategy, memote will be used to detemine which model have the best score. Based on the score and model size, one GSM will be choosen as the model used in this project. 

In [59]:
import numpy as np
from cobra.io import read_sbml_model
from cobra import Reaction, Metabolite

#### Memote report of iMM904.xml

In [60]:
model_iMM904 = read_sbml_model('iMM904.xml')

In [42]:
#This code has been outcommented as it only needed to be runned once. It takes about 10 - 15 min to run this code. 

#%%time
#!memote report snapshot iMM904.xml --filename iMM904.html


From the memote analysis of iMM904.xml, the overall score is 85%.
The consistenty test gav a good result with a sub score of 96%. Here the score of stoichiometric inconsistency was 100% and the mass balance gave a score of 96%. 
For the annotation tests, the score was generlly high (above 80%) except for the gene annotation, which only score 43%

#### Memote report of iND750.xml

In [4]:
#model_iND750 = read_sbml_model('iND750.xml') #This model is not used and is therefore outcommented. 

In [5]:
#This code has been outcommented as it only needed to be runned once. It takes about 10 - 15 min to run this code.

#%%time
#!memote report snapshot iND750.xml --filename iND750.html


The memote analysis of iND750.xml gave a overall score is 86%.
The consistenty test gav a good result with a sub score of 97%. Here the score of stoichiometric inconsistency of 100% and the mass balance gave a score of 97.3%. 
For the annotation tests, the score was generlly high (above 80%) except for the gene annotation, which only score 43%

#### Choice of GSM

Overall the models are fairly similar and both have high scores. However, even though the smallere model iND750 is a bit higher scored, it is the larger model, iMM904, that will be used in this report to similate the production of β-carotene. This is simply because iMM904 contains more genes, reaction and metabolites and therefore is able to simulate the organisme better. Here the different in score is not large engouth to arguee for the use of the smallere model.

*Idea for rewrite:*
From the memote analysis of the two GSM's, the models were overall fairly similar and were both given high scores. 
The smallere model iND750 had a slightly higher score, however, the difference in score is not large enough to arguee for the use of the smallere model. iMM904 contains more genes, reaction and metabolites and is therefore able to simulate the organisme better. Therefore we chose to use the large model iMM904 to simulate the different enigneering strategies for production of β-carotene.

## 4. Computer-Aided Cell Factory Engineering (<1500 words if Category II project)

Since *S. cerevisiae* is not a natural producer of β-carotene, the pathway is not in the model jet. Therefore some new metabolites and reaction must be added to iMM904, so the GSM can be used to predict the production of β-carotene. 

Two pathways are availble for this matter. 

The first pathyway starts at hydroxymethylglutaryl coA, which is produced from acetyl coA naturally in *S. cerevisiae*. In fact it is only the last four reactions in the pathway that needs to be added.

The pathway:
    
    hydroxymethylglutaryl CoA (metabolite id: hmgcoa_c) --> mevalonate (metabolite id: mev_R_c) (Enzyme name: HMG CoA reductase, reaction id: HMGCOAR)

    mevalonate --> --> --> farnesyl diphosphate (metabolite id: frdp_c) 
    
    farnesyl diphosphate --> geranyl diphosphate (metabolite id: ggdp_c) (enzyme name: Farnesyltranstransferase, reaction id: FRTT)

    2 geranyl diphosphate --> Phytoene + 2 diphosphate (metabolite id: ppi_c) (https://www.genome.jp/dbget-bin/www_bget?rn:R10177)

    Phytoene + 3 Acceptors -> Neurosporene + 3 reduced acceptors (https://www.genome.jp/dbget-bin/www_bget?rn:R09691)

    Neurosporene + reduced acceptor + O2 -> Lycopene + acceptor + 2 water (https://www.genome.jp/dbget-bin/www_bget?rn:R04800)

    Lycopene -> beta-Carotene (https://www.genome.jp/dbget-bin/www_bget?rn:R03823)

**Write something about this pathway having unknown acceptors that must be "ignored" or we assume that it is within the enzyme. Argue why we remove it.**

The second pathway **Write something about the second pathway** 



*NB! Nogle links står der accpetor på. Det er en hydrogen acceptor, det vil sige NADP+. den reduceret er NADPH
Husk at loaded model iMM904*

### Metabolites that are usefull and already annotated in the model

In [6]:
#model_iMM904.metabolites.hmgcoa_c
#model_iMM904.metabolites.mev__R_c
#model_iMM904.metabolites.frdp_c
#model_iMM904.metabolites.ggdp_c
#model_iMM904.metabolites.ppi_c
#model_iMM904.metabolites.nadp_c
#model_iMM904.metabolites.nadph_c
#model_iMM904.metabolites.h2o_c
#model_iMM904.metabolites.o2_c
#model_iMM904.reactions.HMGCOAR

### Adding relevant reactions to create beta-carotene pathway:

#### 15-cis-Phytoene syntase
The frist reaction to be added is: 2 Geranylgeranyl diphosphate <=> 15-cis-Phytoene + 2 Diphosphate
(https://www.genome.jp/dbget-bin/www_bget?rn:R10177)
This will allow for Phytoene production which is a precurser in the reaction. 

In [61]:
phy = Metabolite(id='phy', name='phytoene', compartment='c') #Adding the metabolite
new_reaction1 = Reaction('phy_syn') #PHYtoene SYNtase
phy = Metabolite(id='phy', name='phytoene', compartment='c') #Adding the metabolite

new_reaction1.add_metabolites({model_iMM904.metabolites.ggdp_c: -2,
                              phy: 1, # our newly created metabolites
                               model_iMM904.metabolites.ppi_c: 2
                                })
print(new_reaction1.build_reaction_string()) # see if one get the rigth reaction

2 ggdp_c --> phy + 2 ppi_c


The above print statement shows that the reaction was correctly assembled. Below we incorporate the reaction to the GSM and check the occruance. The procedure will be performed on all the reactions added to the model. 

In [62]:
model_iMM904.add_reactions([new_reaction1])
model_iMM904.reactions.phy_syn #check that the reaction is in the model

<Reaction phy_syn at 0x7f4da633ba90>

#### All-trans-Neurosporene Syntase
The next reaction added synthesises all-trans-Neurosporene as seen below: 

15-cis-Phytoene + 3 Acceptor <=> all-trans-Neurosporene + 3 Reduced acceptor
https://www.genome.jp/dbget-bin/www_bget?rn:R09691

In [63]:
new_reaction2 = Reaction('neu_syn') #NEUrosporene SYNtase
#Adding the metabolite
neu = Metabolite(id='neu', name='Neurosporene', compartment='c') 
#AH2 = Metabolite(id='AH2', name='Reduced acceptor', compartment='c') 
#A = Metabolite(id='A', name='Acceptor', compartment='c') 
new_reaction2.add_metabolites({
    model_iMM904.metabolites.nadp_c: -3,
    phy: -1, # our newly created metabolites
    model_iMM904.metabolites.nadph_c: 3,
    neu: 1,
    })
print(new_reaction2.build_reaction_string()) # see if one get the rigth reaction
model_iMM904.add_reactions([new_reaction2])

3 nadp_c + phy --> 3 nadph_c + neu


In [64]:
model_iMM904.reactions.neu_syn #check that the reaction is in the model

<Reaction neu_syn at 0x7f4da633b160>

#### Lycopene synthesis
The production og lycopene is show below: 

all-trans-Neurosporene + Reduced acceptor + Oxygen <=> Lycopene + Acceptor + 2 H2O 

(https://www.genome.jp/dbget-bin/www_bget?rn:R04800)

In [65]:
new_reaction3 = Reaction('lyc_syn') #LYCopene SYNtase
lyc = Metabolite(id='lyc', name='Lycopene', compartment='c') #Adding the metabolite
new_reaction3.add_metabolites({
    neu: -1, # our newly created metabolites
    model_iMM904.metabolites.nadph_c: -1, 
    model_iMM904.metabolites.o2_c: -1,
    lyc: 1,
    model_iMM904.metabolites.nadp_c: 1,
    model_iMM904.metabolites.h2o_c: 2,
    })
print(new_reaction3.build_reaction_string()) # see if one get the rigth reaction

nadph_c + neu + o2_c --> 2 h2o_c + lyc + nadp_c


In [66]:
model_iMM904.add_reactions([new_reaction3])
model_iMM904.reactions.lyc_syn #check that the reaction is in the model

<Reaction lyc_syn at 0x7f4da63315c0>

 #### beta-carotene production
 The last reaction for beta-Carotene production is added:
 
 Lycopene -> beta-Carotene (https://www.genome.jp/dbget-bin/www_bget?rn:R03823)

In [67]:
new_reaction4 = Reaction('beta_syn') #BETA-carotene SYNtase
beta = Metabolite(id='beta', name='Beta-carotene', compartment='c') #Adding the metabolite
new_reaction4.add_metabolites({
    lyc: -1, # our newly created metabolites
    beta: 1,
    })
print(new_reaction4.build_reaction_string()) # see if one get the rigth reaction

lyc --> beta


In [68]:
model_iMM904.add_reactions([new_reaction4])
model_iMM904.reactions.beta_syn #check that the reaction is in the model

<Reaction beta_syn at 0x7f4da633bda0>

#### Beta-carotene extraction from the cell

In [69]:
new_reaction5 = Reaction('EX_beta') #BETA-carotene SYNtase
new_reaction5.add_metabolites({
    beta: -1 #Excretion of beta-carotene
    })
print(new_reaction5.build_reaction_string()) # see if one get the rigth reaction

beta --> 


In [70]:
model_iMM904.add_reactions([new_reaction5])
model_iMM904.reactions.EX_beta #check that the reaction is in the model

<Reaction EX_beta at 0x7f4da633ba20>

### Testing titers of betacarotene production and growth rates

In [71]:
print(model_iMM904.objective)

Maximize
1.0*BIOMASS_SC5_notrace - 1.0*BIOMASS_SC5_notrace_reverse_93090


The growth rate and maximum production of beta-carotene will be calculated in the following: Change this so it will show the growth rate at the hoghest yield and not both of them individually!

In [76]:
with model_iMM904:
    model_iMM904.objective = model_iMM904.reactions.EX_beta 
    print("Maximum beta-carotene flow: ",model_iMM904.optimize().objective_value)
    solution = model_iMM904.optimize()
    print("Growth rate at maximum beta carotene flow: ", solution.fluxes['BIOMASS_SC5_notrace'])

Maximum beta-carotene flow:  0.35143769968051125
Growth rate at maximum beta carotene flow:  0.0


Comment on the growth rate above! - It seams like there is a growth rate of 0.0 when the organism is producing maximum amounts of beta-carotene and here 0.35. LAV PLOT!
We would now like to examine the growth rate  when growing on a different carbon source. 
Therefore we will set the carbon sorce glucose as we examined above as 0 and look at the growth rates at the other medias. Comment on concentration! P:s virker ikke lige C i alle!

In [73]:
#timian will change the code below!
carbon_source_exchanges=[]
for exchange in model_iMM904.exchanges:
    for metabolite in exchange.metabolites:
        if "C" in metabolite.elements:
            print(metabolite.elements)
            carbon_source_exchanges.append(exchange.id)
anaerobic_carbon_sources=[]
for i in range (0,len(carbon_source_exchanges)):
    with model_iMM904:
        medium=model_iMM904.medium
        medium['EX_glc__D_e']=0
        medium[carbon_source_exchanges[i]]=100
        model_iMM904.medium =medium
        solution=model_iMM904.slim_optimize(error_value=0.)
        if solution >0:
            anaerobic_carbon_sources.extend([carbon_source_exchanges[i]])
print(anaerobic_carbon_sources)     

{'C': 1695, 'H': 2995, 'O': 101}
{'C': 28, 'H': 46, 'O': 1}
{'C': 28, 'H': 44, 'O': 1}
{'C': 1695, 'H': 2993, 'O': 101}
{'C': 6, 'H': 10, 'O': 5}
{'C': 2, 'H': 8, 'N': 1, 'O': 1}
{'C': 4, 'H': 7, 'O': 3}
{'C': 2, 'H': 6, 'O': 1}
{'C': 28, 'H': 46, 'O': 1}
{'C': 7, 'H': 14, 'O': 2}
{'C': 5, 'H': 10, 'O': 1}
{'C': 5, 'H': 12, 'O': 1}
{'C': 4, 'H': 8, 'O': 1}
{'C': 8, 'H': 10, 'O': 1}
{'C': 1695, 'H': 2995, 'O': 101}
{'C': 17, 'H': 19, 'N': 4, 'O': 9, 'P': 1}
{'C': 7, 'H': 10, 'O': 5}
{'C': 5, 'H': 10, 'O': 1}
{'C': 1, 'H': 1, 'O': 2}
{'C': 6, 'H': 12, 'O': 6}
{'C': 6, 'H': 9, 'O': 3}
{'C': 4, 'H': 9, 'N': 1, 'O': 2}
{'C': 4, 'H': 2, 'O': 4}
{'C': 8, 'H': 20, 'N': 1, 'O': 6, 'P': 1}
{'C': 7, 'H': 6, 'N': 1, 'O': 2}
{'C': 5, 'H': 9, 'N': 1, 'O': 3}
{'C': 9, 'H': 18, 'O': 11, 'P': 1}
{'C': 6, 'H': 12, 'O': 6}
{'C': 9, 'H': 17, 'N': 1, 'O': 3}
{'C': 20, 'H': 22, 'N': 2, 'O': 8}
{'C': 5, 'H': 12, 'O': 5}
{'C': 2, 'H': 3, 'O': 2}
{'C': 2, 'H': 4, 'O': 1}
{'C': 4, 'H': 8, 'O': 2}
{'C': 6, 'H': 

#### Improving the yield of beta-caroten through knockouts


Reading ..(reference) different knowck strdegies was tried to improve the flow of beta caroten.
First we try to knock out Glyceraldehyde-3-phosphate dehydrogenase:

In [56]:
model_iMM904.reactions.GAPD

<Reaction GAPD at 0x7f4da73e20f0>

In [78]:
with model_iMM904:
    model_iMM904.reactions.GAPD.bounds = 0,0
    model_iMM904.objective = model_iMM904.reactions.EX_beta 
    print("Maximum beta-carotene flow: ",model_iMM904.optimize().objective_value) 

Maximum beta-carotene flow:  0.05714285714285716


Here it can be seen that the flow of 0.0571 of beta caroten is lower than before where the flow was 0.351.
Next we try to know out Pyruvate kinase:

In [19]:
model_iMM904.reactions.PYK

<Reaction PYK at 0x7f4dabfc9588>

In [20]:
with model_iMM904:
    model_iMM904.reactions.PYK.bounds = 0,0
    model_iMM904.objective = model_iMM904.reactions.EX_beta 
    print("Maximum beta-carotene flow: ",model_iMM904.optimize().objective_value)

Maximum beta-carotene yield:  0.07334963325183375


Here it can be seen that the flow of 0.0733 of beta caroten again is lower than before where the flow was 0.351.
Next we try to know out Arginase:

In [21]:
model_iMM904.reactions.ARGN

<Reaction ARGN at 0x7f4dacbe0198>

In [22]:
with model_iMM904:
    model_iMM904.reactions.ARGN.bounds = 0,0
    model_iMM904.objective = model_iMM904.reactions.EX_beta 
    print("Maximum beta-carotene flow: ",model_iMM904.optimize().objective_value)

Maximum beta-carotene yield:  0.351437699680511


Here it can be seen that the flow of 0.351 is the same as before we started knowcking out genes. 
Next we try to know out Pyruvate decarboxylase:

In [80]:
model_iMM904.reactions.PYRDC

<Reaction PYRDC at 0x7f4da5df09e8>

In [86]:
with model_iMM904:
    model_iMM904.reactions.PYRDC.bounds = 0,0
    model_iMM904.objective = model_iMM904.reactions.EX_beta 
    print("Maximum beta-carotene flow: ",model_iMM904.optimize().objective_value)

Maximum beta-carotene flow:  0.34546386468477747


Here it can be seen that the flow is 0.34 and slightly lower than to be begin with. 
Next we try to knockout different combinations of genes to see if this have an impact on the flow of beta carotene. 

In [92]:
with model_iMM904:
    model_iMM904.reactions.PYRDC.bounds = 0,0
    model_iMM904.reactions.ARGN.bounds = 0,0
    model_iMM904.reactions.GAPD.bounds = 0,0
    model_iMM904.reactions.PYK.bounds = 0,0
    model_iMM904.reactions.ATPS.bounds = 0,0
    model_iMM904.objective = model_iMM904.reactions.EX_beta 
    print("Maximum beta-carotene flow: ",model_iMM904.optimize().objective_value)

Maximum beta-carotene flow:  0.057142857142857106


Different combinations of gene knockout did not affect the final flow of beta-carotene. 

## 5. Discussion (<500 words)

## 6. Conclusion (<200 words)

## References

[1] Madigan, M. T., Martinko, J. M., & Parker, J. (2014). Brock biology of microorganisms. Upper Saddle River, NJ: Prentice Hall/Pearson Education.

[2] Qian Li, Zhiqiang Sun, Jing Li, Yansheng Zhang, Enhancing beta-carotene production in Saccharomyces cerevisiae by metabolic engineering, FEMS Microbiology Letters, Volume 345, Issue 2, August 2013, Pages 94–101, https://doi.org/10.1111/1574-6968.12187